src/store/modules/layers.store.js

```diff
@@ -135,27 +122,81 @@ const getters = {
         const layer = state.activeLayers.find((layer) => layer.getID() === layerId)
         return layer?.isExternal || (layer?.type === LayerTypes.KML && !layer?.adminId)
     },
+    /**
+     * Returns the z-index of a visible layer, taking into account the background layer(s) and/or
+     * if one of them is a group of layer (and thus all sub-layers are counted in the index)
+     *
+     * If the layer is not part of the visible layers (or is null or invalid), this will return -1 as a result
+     *
+     * @param {AbstractLayer} layerIdOrObject
+     * @returns {Number}
+     */
+    zIndexForVisibleLayer: (state, getters) => (layerIdOrObject) => {
+        let lookupId
+        if (layerIdOrObject instanceof AbstractLayer) {
+            lookupId = layerIdOrObject.getID()
+        } else {
+            lookupId = layerIdOrObject
+        }
```

Comment:
Might be wise to check for string type here and in case it is not a string to log an error, for debuging might be useful